internal/object/object.go

```diff
@@ -117,46 +126,30 @@ func (o *Object) UpdateMetadata(source int64, name string, url types.String, ext
 	}
 
 	stmt, _ := o.db.BuildUpsertStmt(&SourceMetadata{})
-	_, err := o.db.NamedExec(stmt, sourceMetadata)
+	_, err := tx.NamedExecContext(ctx, stmt, sourceMetadata)
 	if err != nil {
-		return fmt.Errorf("failed to upsert object metadata: %s", err)
+		return err
 	}
 
-	tx, err := o.db.BeginTxx(context.TODO(), nil)
-	if err != nil {
-		return fmt.Errorf("failed to start transaction for object extra tags: %s", err)
-	}
-	defer tx.Rollback()
-
 	extraTag := &ExtraTagRow{ObjectId: o.ID, SourceId: source}
-	_, err = tx.NamedExec(`DELETE FROM "object_extra_tag" WHERE "object_id" = :object_id AND "source_id" = :source_id`, extraTag)
+	_, err = tx.NamedExecContext(ctx, `DELETE FROM "object_extra_tag" WHERE "object_id" = :object_id AND "source_id" = :source_id`, extraTag)
 	if err != nil {
-		return fmt.Errorf("failed to delete object extra tags: %s", err)
+		return err
 	}
 
 	if len(extraTags) > 0 {
 		stmt, _ = o.db.BuildInsertStmt(extraTag)
-		_, err = tx.NamedExec(stmt, sourceMetadata.mapToExtraTags())
+		_, err = tx.NamedExecContext(ctx, stmt, sourceMetadata.mapToExtraTags())
 		if err != nil {
-			return fmt.Errorf("failed to insert object extra tags: %s", err)
+			return err
 		}
 	}
 
-	if err = tx.Commit(); err != nil {
-		return fmt.Errorf("failed to commit object extrag tags transaction: %s", err)
-	}
-
 	if o.Metadata == nil {
 		o.Metadata = make(map[int64]*SourceMetadata)
 	}
 
-	if m := o.Metadata[source]; m != nil {
-		m.Name = name
-		m.URL = url
-		m.ExtraTags = extraTags
-	} else {
-		o.Metadata[source] = sourceMetadata
-	}
+	o.Metadata[source] = sourceMetadata
```

Comment:
Note to my self: double-check that there are no assumptions on constant addresses in the code. (This clearly looks like code I wrote to keep the pointer in `o.Metadata[source]` constant, at least for the config, there were some assumptions on that.)